shared_model/interfaces/iroha_internal/transaction_batch_factory.hpp

```diff
@@ -33,7 +33,8 @@ namespace shared_model {
        */
       template <typename TransactionValidator,
                 typename FieldValidator = validation::FieldValidator>
-      static iroha::expected::Result<TransactionBatch, std::string>
+      static iroha::expected::Result<std::unique_ptr<TransactionBatch>,
```

Comment:
Unsafe factory should be removed, and this factory should be refactored with polymorphism, similar to existing ProposalFactory.